src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -390,14 +389,62 @@
 
             if (validIssuerTemplate.Contains(TenantIdTemplate))
             {
-                return validIssuerTemplate.Replace(TenantIdTemplate, tenantId) == actualIssuer;
+                return IssuersWithTemplatesAreEqual(validIssuerTemplate, TenantIdTemplate, actualIssuer, tenantId);
             }
             else
             {
                 return validIssuerTemplate == actualIssuer;
             }
         }
 
+        /// <summary>
+        /// Compare two Issuers with templates without string allocations.
+        /// </summary>
+        internal static bool IssuersWithTemplatesAreEqual(string issuer1, string issuer1Template, string issuer2, string issuer2Template)
+        {
+            int firstHalfIssuer1EndIndex = issuer1.IndexOf(issuer1Template, StringComparison.Ordinal);
+            int firstHalfIssuer2EndIndex = issuer2.IndexOf(issuer2Template, StringComparison.Ordinal);
+
+            // ensure indices exist for both issuers
+            if (firstHalfIssuer1EndIndex == -1  || firstHalfIssuer2EndIndex == -1)
+                return false;
+
+            // if the indices aren't equal, then the issuers aren't equal
+            if (firstHalfIssuer1EndIndex != firstHalfIssuer2EndIndex)
+                return false;
+
+            // ensure the first part of the issuer1 matches the first part of issuer2
+            if (!StringSegmentsAreEqual(issuer1, 0, firstHalfIssuer1EndIndex, issuer2, 0, firstHalfIssuer2EndIndex))
```

Comment:
You can save the allocation above (line 405-406, 420-421) by checking that the issuer1Template and issuer2Template length is the same at the very beginning of the function.

Reply:
Great, done